wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -576,20 +580,36 @@ private String sessionUserNameOr(String nameIfNoUser) {
     return nameIfNoUser;
   }
 
-  private class CleanupRuntimeOnAbnormalRuntimeStop implements EventSubscriber<RuntimeStatusEvent> {
+  private class AbnormalRuntimeStopListener implements EventSubscriber<RuntimeStatusEvent> {
     @Override
     public void onEvent(RuntimeStatusEvent event) {
       if (event.isFailed()) {
         RuntimeState state = runtimes.remove(event.getIdentity().getWorkspaceId());
         if (state != null) {
+          String workspaceId = state.runtime.getContext().getIdentity().getWorkspaceId();
```

Comment:
2 lines up workspace ID is fetched too. Can you move this assignment higher and reuse the result?